### ISSUE TITLE
A bug occurs on update

### ISSUE THREAD

**Issue by asmsuechan** [BoostIO/Boostnote#243]
Hi, I hit a bug on update...

<img width="532" alt="2017-01-16 1 09 57" src="https://cloud.githubusercontent.com/assets/11307908/21964088/fbe77474-db88-11e6-9909-df07d9c618a1.png">

Does anyone arise it?

After I had killed Boostnote process and restarted, it was updated properly (v0.8.2).

**Comment by kazup01** [comment]
We are investigating the cause of this bug.
Downloading the latest version from the official page(http://boostnote.io/#download) will work fine, so I recommend it.
Of course the datas are not goes out, it is possible to take over without problems.

**Comment by sota1235** [comment]
Thank you for reporting.
What is the version you used? Please tell me :bow:

**Comment by asmsuechan** [comment]
It occurs on update from v0.8.1 to v0.8.2.

**Comment by sota1235** [comment]
I see. I try to investigate...!

**Comment by asmsuechan** [comment]
I blamed the line of the bug, I recognize it is my fault :fearful:
https://github.com/BoostIO/Boostnote/blob/master/lib/main-window.js#L52

Perhaps `if (mainWindow != null)` is necessary.

**Comment by asmsuechan** [comment]
electron-gh-releases is used on update (https://github.com/BoostIO/Boostnote/blob/master/lib/main-app.js#L79)

And `install()` method uses `quitAndInstall()`, according to the reference of `quitAndInstall()` ([here](https://github.com/electron/electron/blob/master/docs/api/auto-updater.md#autoupdaterquitandinstall)),
> Note: autoUpdater.quitAndInstall() will close all application windows first and only emit before-quit event on app after that. This is different from the normal quit event sequence.

The bug occurs at `before-quit` event. Thus it's related with the bug.

**Comment by asmsuechan** [comment]
Thus I tried to create `quit` event like below however same error still occur.
```js
  app.on('quit', function (e) {
    config.set('windowsize', mainWindow.getBounds())
  })
```

**Comment by sota1235** [comment]
Great fixing by @asmsuechan !!
It will be released on v0.8.3

**Comment by kazup01** [comment]
Thank you for your help, @asmsuechan !👏

**Comment by sota1235** [comment]
It's fixed on v0.8.3. Thank you for reporting!
